tests/epics/motion/test_motor.py

```diff
@@ -18,23 +21,38 @@ async def sim_motor():
     async with DeviceCollector(sim=True):
         sim_motor = motor.Motor("BLxxI-MO-TABLE-01:X")
         # Signals connected here
+    AsyncStatus.wrap
 
+    async def fake_stop(*args, **kwargs):
+        sim_motor.motor_done_move._backend._set_value(True)  # type: ignore
+        await asyncio.sleep(0.01)
+
+    sim_motor.motor_stop.trigger = fake_stop  # type: ignore
     assert sim_motor.name == "sim_motor"
     set_sim_value(sim_motor.motor_egu, "mm")
     set_sim_value(sim_motor.precision, 3)
     set_sim_value(sim_motor.velocity, 1)
+    set_sim_value(sim_motor.motor_done_move, True)
     yield sim_motor
 
 
+async def wait_for_eq(item, attribute, comparison, timeout):
+    timeout_time = time.monotonic() + timeout
+    while getattr(item, attribute) != comparison:
+        await asyncio.sleep(A_BIT)
+        if time.monotonic() > timeout_time:
+            raise TimeoutError
+
+
 async def test_motor_moving_well(sim_motor: motor.Motor) -> None:
```

Comment:
Please could you revert this test and check it still passes?

Reply:
It doesn't the way it is in main - it needs a sleep after the `set_mock_value(readback ...)` to process. This slight alteration does pass:



```python
async def test_motor_moving_well_2(sim_motor: motor.Motor) -> None:
    set_mock_put_proceeds(sim_motor.user_setpoint, False)
    s = sim_motor.set(0.55)
    watcher = Mock()
    s.watch(watcher)
    done = Mock()
    s.add_callback(done)
    await asyncio.sleep(A_BIT)
    assert watcher.call_count == 1
    assert watcher.call_args == call(
        name="sim_motor",
        current=0.0,
        initial=0.0,
        target=0.55,
        unit="mm",
        precision=3,
        time_elapsed=pytest.approx(0.0, abs=0.05),
    )
    watcher.reset_mock()
    assert 0.55 == await sim_motor.user_setpoint.get_value()
    assert not s.done
    await asyncio.sleep(0.1)
    set_mock_value(sim_motor.user_readback, 0.1)
    await asyncio.sleep(0.1)
    assert watcher.call_count == 1
    assert watcher.call_args == call(
        name="sim_motor",
        current=0.1,
        initial=0.0,
        target=0.55,
        unit="mm",
        precision=3,
        time_elapsed=pytest.approx(0.11, abs=0.05),
    )
    set_mock_put_proceeds(sim_motor.user_setpoint, True)
    await asyncio.sleep(A_BIT)
    assert s.done
    done.assert_called_once_with(s)
```